lib/devise/secret_key_finder.rb

```diff
@@ -7,14 +7,12 @@ def initialize(application)
     end
 
     def find
-      if @application.respond_to?(:credentials) && key_exists?(@application.credentials)
-        @application.credentials.secret_key_base
+      if @application.respond_to?(:secret_key_base) && key_exists?(@application)
```

Comment:
```suggestion
      if @application.respond_to?(:secret_key_base)
```

I think if Rails version is recent enough to respond here, the method is guaranteed to return a value or raise.